src/jobs/ega/axios/egaClient.ts

```diff
@@ -179,30 +123,72 @@ export const egaApiClient = async () => {
   apiAxiosClient.interceptors.response.use(
     (response) => response,
     async (error) => {
-      if (error.response && error.config) {
-        switch (error.response.status) {
-          case 401:
-            if (!refreshTokenPromise) {
-              resetAccessToken();
+      if (error instanceof AxiosError) {
+        // Must check for error.response *before* error.request, because a 401 error will also trigger error.request
+        // This can cause an endless loop where the token is never refreshed
+        if (error.response) {
+          if (error.config) {
+            logger.error(`${CLIENT_NAME} - AxiosError - error.response - error.config`);
+            switch (error.response.status) {
+              case 401:
+                logger.info('Access token expired');
+                if (!refreshTokenPromise) {
+                  resetAccessToken();
+                }
+                const updatedAccessToken = await getAccessToken();
+                const refreshedBearerToken = `Bearer ${updatedAccessToken.access_token}`;
+                // set new token on original request that had the 401 error
+                error.config.headers['Authorization'] = refreshedBearerToken;
+                // reset on client headers so subsequent requests have new access token
+                apiAxiosClient.defaults.headers['Authorization'] = refreshedBearerToken;
+                // returns Promise for original request
+                return apiAxiosClient.request(error.config);
+              case 400:
+                // don't retry
+                logger.error(`Bad Request`);
+                return new BadRequestError(error.message);
+              case 404:
+                logger.error(`Not Found`);
+                // don't retry
+                return new NotFoundError(error.message);
+              case 429:
+                logger.error(`Too Many Requests`);
+                logger.error(
+                  `${CLIENT_NAME} - ${error.response.status} - ${error.response.statusText} - retrying original request.`,
+                );
+                // retry original request. this response error shouldn't be an issue because throttling is in place
+                return apiAxiosClient.request(error.config);
```

Comment:
`apiAxiosClient.request` is not throttled. This request can occur in addition to all the throttled requests, no?

This holds true for every use of client.request(). Should we throw a wrapper over that to throttle it?